Quick starter (#4)
shopware/admin-api-reference: docs/guides/Quick Start/03-orders.md

      "type": "string"
    },
    "languageId": {
      "description": "Unique ID of language",
      "type": "string"
    },
    "salesChannelId": {
      "description": "Unique ID of defined sales channel",
      "type": "string"
    },
    "orderDateTime": {
      "description": "Timestamp of the order placed",
      "type": "string"
    },
    "currencyFactor": {
      "description": "Rate factor for currencies",
      "type": "string"
    },
    "stateId": {
      "description": "Unique ID of transition state as defined by state machine",
      "type": "string"
      }
  }
}
```

## Order line item

An order can be have more other items or child items of `type` : `product`, `promotion`, `credit` or `custom`. To fetch line items for a particular order, try the below API request:

```json http
{
  "method": "get",
  "url": "http://localhost/api/order/558efc15fe604829b4d0607df75187e0/line-item",
  "headers": {
    "Content-Type": "application/json",
    "Accept": "application/json",
    "Authorization": "Bearer Your_API_Key"
  }
  }
```

Every order created is associted with order, payment, and delivery transitions. More details are mentioned below:

> ## Order transitions

Reviewer: ```suggestion
## Order state handling
```

```json http
{
  "method": "post",
  "url": "http://localhost/api/_action/order_delivery/558efc15fe604829b4d0607df75187e0/state/fail",
  "headers": {
    "Content-Type": "application/json",
    "Accept": "application/json",
    "Authorization": "Bearer Your_API_Key"
  }
}
```

### Order transaction

The order transaction state represents the state of the payment. `open`, `fail`, `authorize`, `refund_partially`, `refund`, `do_pay`, `paid`, `paid_partially`, `remind`, `cancel` are the states associated with order transaction.

```json http
{
  "method": "post",
  "url": "http://localhost/api/_action/order_transaction/558efc15fe604829b4d0607df75187e0/state/open",
  "headers": {
    "Content-Type": "application/json",
    "Accept": "application/json",
    "Authorization": "Bearer Your_API_Key"
  }
  }
```  

## Refund Payment

[Initiating and capturing payment is handled by store-api](https://shopware.stoplight.io/docs/store-api/8218801e50fe5-handling-the-payment), whereas payment refund is dealt by the admin-api.

Refund payment method can be called only for transactions that are claimed to be successful. The payment can either be completely or partially refunded.

> Generally, refunds are linked to a specific transaction ID or order ID. A refund can have multiple positions, with different order line items and amounts. To allow easy refund handling, have your payment handler implement the `RefundPaymentHandlerInterface`. This validates the legitimacy of the refund and call the PSP to refund the given transaction.

Reviewer 2: ```suggestion
Generally, refunds are linked to a specific order transaction capture ID. A refund can have multiple positions, with different order line items and amounts. To allow easy refund handling, have your payment handler implement the `RefundPaymentHandlerInterface`. Your RefundHandler implementation will be called with the `refundId` where you can call you PSP or similar.
```

> Use the Refund API `/api/_action/order_transaction_capture_refund/{refundId}` endpoint to refund payments. You usually need the `refundId` and `context`. The refundId is the id of the `OrderTransactionCaptureRefund` entity, which the payment plugin has created before.

Reviewer 2: ```suggestion
Use the Refund API `/api/_action/order_transaction_capture_refund/{refundId}` endpoint to refund payments. You usually need the `refundId` and `context`. The refundId is the id of the `OrderTransactionCaptureRefund` entity, which the payment plugin has created before, and will be forwarded to your RefundHandler implementation.
```

> When you refund a payment, the Refunds API creates an object that provides refund details — the information on the amount, the referenced capture and, if provided, a reason and specific positions which items are being refunded. Once the refund is successfull, the `stateHandler` changes the state to *complete*.

Reviewer 2: ```suggestion
When you refund a payment, the API will change the refund state to *complete*. If you want to fail the refund in your RefundHandler implementation, simply throw a `RefundException`, and the state of the refund will transition to *fail*. 
```

> In case of refund failures, `RefundException` will handle it.

Author: @lernhart - Please review this info and suggest any changes if required.

> # Order Overview

Reviewer: I'm not what it indicates, if there's a top level heading that says "overview" without any lower level documents.

The order resource enables merchants to process orders they receive from customers.

You can use the orders resource to do the following:

* Retrieve orders and their line-items

> * Create dummy orders 

Reviewer: I would not call it "dummy" orders, but illustrate in the corresponding section what types of orders you can create.

* Update order statuses
* Manage returns

## Retrieve Orders

> A list of all customer orders is obtained using the route:

Reviewer: ```suggestion
A list of all customer orders is obtained using the `/api/search/order` route.

> ``` markdown
> POST /api/search/order
> ```

Reviewer: ```suggestion
```json http
{
  "method": "post",
  "url": "/api/search/order",
  "headers": {
    "Content-Type": "application/json",
    "Accept": "application/json",
    "Authorization": "Bearer Your_API_Key"
  }
}
```
```

Reviewer: You can also mention, that sorting, aggregating and narrowing down order results is possible using a search criteria and referencing the corresponding section about [search queries](https://shopware.stoplight.io/docs/store-api/cf710bf73d0cd-search-queries).

### Order line-items

An order can have more other items or child items of `type` : `product`, `promotion`, `credit` or `custom`. To fetch line items for a particular order, try the below route:

> ```text
> GET /api/order/{order-id}/line-item
> ```
> Sample API request:
> 

Reviewer: ```suggestion
```

```json http
{
  "method": "get",
  "url": "http://localhost/api/order/558efc15fe604829b4d0607df75187e0/line-item",
  "headers": {
    "Content-Type": "application/json",
    "Accept": "application/json",
    "Authorization": "Bearer Your_API_Key"
  }
  }
```

## Create an order

> You can create orders manually in the admin panel to record orders made outside of Shopware or send customer invoices.

Reviewer: In here maybe explain the limitation of creating orders that way - e.g. no payment attached, no invoices are created automatically etc.

    "orderDateTime": "2021-05-31T14:13:14.866+00:00",
    "currencyFactor": 1.0,
    "stateId": "a75eb89b4abe41f9bade83b2f07d874e"
   }
  }
}
``` 

```json json_schema
{
  "type": "object",
  "description": "Parameters for order creation",
  "properties": {
    "billingAddressId": {
      "description": "ID of the billing address",
      "type": "string"
    },
    "currencyId": {
      "description": "ID of the currency to which the price belongs",
      "type": "string"
    },
    "languageId": {
      "description": "Unique ID of language",
      "type": "string"
    },
    "salesChannelId": {
      "description": "Unique ID of defined sales channel",
      "type": "string"
    },
    "orderDateTime": {
      "description": "Timestamp of the order placed",
      "type": "string"
    },
    "currencyFactor": {
      "description": "Rate factor for currencies",
      "type": "string"
    },
    "stateId": {
      "description": "Unique ID of transition state as defined by state machine",
      "type": "string"
      }
  }
}
```

Orders created are associated with payment and delivery transitions. The following section provides you with more details.

## Order state handling

> Every order in Shopware has three state machines, `order.state`, `order_delivery.state`, `order_transaction.state`, that holds the status of an order, delivery, and payment status, respectively. The `state_machine_transition` is a collection of all defined transitions defined.

Reviewer: ```suggestion
Every order in Shopware has three states `order.state`, `order_delivery.state`, `order_transaction.state`. For each state there is a state machine that holds the status of the order, delivery, and payment status, respectively. The `state_machine_transition` is a collection of all defined transitions.
```

### Order delivery

The order delivery state represents the state of the delivery. `reopen`, `ship`, `ship_partially`, `cancel`, `retour`, and `retour_partially` are the states associated with order delivery.

Below is a sample request that sets the delivery state to *fail*:

```json http
{
  "method": "post",
  "url": "http://localhost/api/_action/order_delivery/558efc15fe604829b4d0607df75187e0/state/fail",
  "headers": {
    "Content-Type": "application/json",
    "Accept": "application/json",
    "Authorization": "Bearer Your_API_Key"
  }
}
```

### Order transaction

The order transaction state represents the state of the payment. `open`, `fail`, `authorize`, `refund_partially`, `refund`, `do_pay`, `paid`, `paid_partially`, `remind`,  and `cancel` are the states associated with order transaction.

Below is a sample request that sets the payment state to *open*:

```json http
{
  "method": "post",
  "url": "http://localhost/api/_action/order_transaction/558efc15fe604829b4d0607df75187e0/state/open",
  "headers": {
    "Content-Type": "application/json",
    "Accept": "application/json",
    "Authorization": "Bearer Your_API_Key"
  }
  }
```  

## Refund Payment

> [Initiating and capturing payment is handled by store-api](https://shopware.stoplight.io/docs/store-api/8218801e50fe5-handling-the-payment), whereas the admin API deals with refund payment.

Reviewer: ```suggestion
Initiating and capturing payment is handled by [Store API](https://shopware.stoplight.io/docs/store-api/8218801e50fe5-handling-the-payment), whereas the admin API deals with refund payment.
```

The refund payment method can be called only for transactions that are claimed to be successful. 

> Generally, refunds are linked to a specific order transaction capture ID. An order can have one or more line items and amounts. Each of these line items signifies refund positions. Based on the number of line items requested for refund, the payment can either be partially or wholly refunded.

Reviewer: Matter of taste I think - you decide :)

```suggestion
Generally, refunds are linked to a specific order transaction capture ID. An order can have one or more line items and amounts. Each of these line items signifies refund positions. Based on the number of line items requested for refund, the payment can either be partially or completely refunded.
```

> To allow easy refund handling, have your payment handler implement the `RefundPaymentHandlerInterface`. Your RefundHandler implementation will be called with the `refundId` to call your PSP or similar.
> 
> Use the Refund API endpoint `/api/_action/order_transaction_capture_refund/{refundId}`  to refund payments. You usually need the `refundId` and `context`. The refundId is the id of the `OrderTransactionCaptureRefund` entity, which the payment plugin has created before and will be forwarded to your RefundHandler implementation.

Reviewer: This requires in-depth knowledge about Shopware Extension (Plugin) Development, so we should mark it somehow (maybe with a warning) and redirect to resources for extension developers. 

```suggestion
<!-- theme: danger -->

> **Refund handling internals** 
>
> To allow easy refund handling, have your payment handler implement the `RefundPaymentHandlerInterface`. Your refund handler implementation will be called with the `refundId` to call the PSP or gateway.
>
> For more information check our documentation on [implementing payment refund handlers](https://developer.shopware.com/docs/guides/plugins/plugins/checkout/payment/add-payment-plugin#refund-example)
```

![image](https://user-images.githubusercontent.com/7780750/203100234-3dad675d-df6c-4b06-865a-8f852cabe624.png)
